fc/cli/reports.py

```diff
@@ -0,0 +1,267 @@
+import click
+from ..auth.combo import ComboAuth
+from ..jira import tasks
+from . import cli_library
+from google.oauth2 import service_account
+from googleapiclient.discovery import build
+import requests
+from typing import Sequence
+from datetime import date
+
+
+@click.group()
+def reports():
+    pass
+
+
+@reports.command()
+@click.option('--username', )
+def usertasks(username: str):
+    cli_library.echo('Generating user tasks to google sheet')
+
+    auth = ComboAuth(username)
+
+    # setup and build credentials for google api calls
+    SCOPES = ['https://www.googleapis.com/auth/spreadsheets', 'https://www.googleapis.com/auth/drive.file']
+    SERVICE_ACCOUNT_FILE = auth.google_service_acct_creds()
+    SHEET_CREATE_URL = auth.sheet_create_url()
+
+    if SERVICE_ACCOUNT_FILE is None or SHEET_CREATE_URL is None:
+        cli_library.echo('Google service account credential file path and sheet create url must be defined '
+                         'in order to generate a report file.')
+
+    credentials = service_account.Credentials.from_service_account_file(
+        SERVICE_ACCOUNT_FILE, scopes=SCOPES)
+
+    service = build('sheets', 'v4', credentials=credentials)
+
+    report_title = 'User Task Report ' + date.today().strftime('%m-%d-%y')
+
+    response = requests.get(SHEET_CREATE_URL + report_title)
+    response.raise_for_status()
+
+    response_json = response.json()
+
+    spreadsheet_id = None
+    folder_id = None
+
+    if response_json['success']:
+        spreadsheet_id = response_json['newSheetId']
+        folder_id = response_json['destFolderId']
+        cli_library.echo('successful spreadsheet creation, https://docs.google.com/spreadsheets/d/{}/edit#gid=0, '
+                         .format(spreadsheet_id) +
+                         'folder: {}'.format(folder_id))
+
+    # update the individual sheet name in the spreadsheet we just created
+    sheet_update = {
+        'requests': {
+            'updateSheetProperties': {
+                'properties': {
+                    'title': 'Names'
+                },
+                'fields': 'title'
+            }
+        }
+    }
+
+    response = service.spreadsheets().batchUpdate(
+        spreadsheetId=spreadsheet_id,
+        body=sheet_update).execute()
+
+    cli_library.echo('first sheet title updated')
+
+    # add data header row for names sheet
+    # get users
+    # iterate through users
+    #   add data row to name sheet with name
+    #   create an add sheet request for the user
+    #   create a data header row for user sheet
+    #   get tasks for user
+    #     add data row to user sheet for each task: key (link), issue type, summary, status, vfr, due date, importance
+    #     , loe
+    # execute batch update for adding all sheets
+    # execute batch update for adding all data
```

Comment:
A lot of the following code seems like it can be split up into separate chunks, aka functions.  Can we split it into functions, please?